frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import * as _ from 'lodash';
+import {
+  Alert,
+  DataList,
+  DataListCheck,
+  DataListItem,
+  DataListItemRow,
+  DataListCell,
+  DataListItemCells,
+} from '@patternfly/react-core';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { setColumnManagementFilter } from '../../actions/ui';
+
+export const MAX_VIEW_COLS = 9;
+
+const OVERRIDES = ['Name'];
+
+const getDataListRows = (
+  rowData: any[],
+  onChange: (checked: boolean, event: any) => void,
+  disableUncheckedRows: boolean,
+): React.ReactNode[] =>
+  rowData.map((column) => (
+    <DataListItem aria-labelledby="table-column-management-item1" key={column.title}>
+      <DataListItemRow>
+        <DataListCheck
+          isDisabled={(disableUncheckedRows && !column.visible) || OVERRIDES.includes(column.title)}
+          aria-labelledby="table-column-management-item1"
+          checked={column.visible}
+          name={column.title}
+          onChange={onChange}
+        />
+        <DataListItemCells
+          dataListCells={[
+            <DataListCell id={`table-column-management-item-${column.title}`} key={column.title}>
+              {column.title}
+            </DataListCell>,
+          ]}
+        />
+      </DataListItemRow>
+    </DataListItem>
+  ));
+
+export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
+  kinds,
+  cancel,
+  close,
+  columnFilters,
+}) => {
+  const dispatch = useDispatch();
+  const initialDefaultColumns = !_.isEmpty(columnFilters)
+    ? _.cloneDeep(columnFilters).filter((column) => column.title.length > 0 && !column.additional)
+    : {};
+  const initialAdditionalColumns = !_.isEmpty(columnFilters)
+    ? _.cloneDeep(columnFilters).filter((column) => column.additional)
+    : {};
+  const [defaultColumns, setDefaultColumns] = React.useState<any[]>(initialDefaultColumns);
+
+  const [additionalColumns, setAdditionalColumns] = React.useState<ColumnHeaderRow[]>(
+    initialAdditionalColumns,
+  );
+
+  const updateColumns = (checked: boolean, name: string, columnArray: any): any => {
```

Comment:
Use a better type than `any`.

Reply:
I'd suggest storing checked state internally as a map and only converting to an array when saved. Then you don't need to complicated array manipulation on every change.

Reply:
@spadgett not sure how the map will help make things less complex here...also not sure its super complicated...i tried converting, but it seemed to make things more difficult. Maybe i'm missing something here.